### DISCUSSION TITLE
react-navigaton useNavigation is causing my app to crash

### DISCUSSION THREAD

**Discussion by AlexMachin1997** [react-navigation/react-navigation#9816] [Q&A] [answered]
Hi everyone,

So I'm in the process of rewriting my old react-navigation app with the new Expo SDK (v42). So far I've rewritten all the screens with hooks, however I've come to get the navigation portion working and I'm having some difficulties which is why I'm here.

I've not too much experience with react-navigation, I come from a react-router background but some concepts overlap. So from what I remember as long as you wrap the application in a Router Context (NavigationContainer for this library) an you call the useNavigation you can access the various navigation related properties and functions.

The problem I'm having is when I call the navigate() which is provided by the useNavigation hook my application just closes down, no logs are printed no stack trace appears to indicate the problem. I might be using it incorrectly as I've based it on what I had previosuly which in theory should work though the api might have changed and I could be using it incorrectly.

I've not commited the code as it's not finished (I can commit it if needed), heres the general navigation flow of my app.

**My applications current navigation flow:**

- WelcomeStack, this is just a tutorial. Once the "done" button is clickedit updates my apps UserProfile Context and it switches to the DrawerStack.

- The DrawerStack then renders a navigation drawer and the MainStack

- The MainStack contains the TabStack, a single item page and a settings page

If anyone could provide some guidance it would greatly be appricated.

**Comment by raajnadar** [comment]
Can you provide a snack.expo.io repro?

**Reply by AlexMachin1997** [reply]
Hi @raajnadar ,

I wasn't able to create a snack, but I did end up commiting the code I currently have for the proejct.

The code can be found [here](https://github.com/AlexMachin1997/Shopping-List-Assistant-With-Expo-Rewrite)

Sorry if it's insufficent, I would create a snack but I was unable to it kept erroring when I tried important the git repo.

**Reply by raajnadar** [reply]
No need to add the whole codebase add only the code where the error is thrown. Or the screen which is causing the crash.

**Reply by AlexMachin1997** [reply]
Oh my bad, first time posting in a Github discussion 😄 

This is the file which is causing the issue, it can be found [here ](https://github.com/AlexMachin1997/Shopping-List-Assistant-With-Expo-Rewrite/blob/master/src/screens/ShoppingLists.js#L155).  It's the navigation.navigate() part is when the error occurs, when an item is clicked to go to the single item screen the app closes down. I did try navigating in various other parts of the app and It's the same problem.

**Reply by raajnadar** [reply]
Can you try `navigation.navigate('ShoppingList')` without any params? And let me know what happens

**Reply by raajnadar** [reply]
went through the package.json & found you have react navigation v4 please remove that https://stackoverflow.com/a/68638223/5519872, clear the metro cache and try again.

**Reply by AlexMachin1997** [reply]
@raajnadar Sorry for the late response, I didn't get notified about the comment or I missed it in my inbox.

Thank's for checking my package.json, hopefully that clears things up. I'll give that a go at the weekend see what happens.. I'll  report back then.

**Reply by AlexMachin1997** [reply]
@raajnadar no luck yet. I've tried the following:

- Clearing the Expo cache
- Removed my yarn.lock
- Installed all the dev and none dev dependencies
- I've tried navigating to the page without params but nothing, I got the same behaviour as before.

I can see a warning in the console, but it's related to how I'm importing my components (See image below)
![image](https://user-images.githubusercontent.com/29917195/131361534-2b95915a-e9c1-47e0-9425-5dc5834b2c2f.png)

I'm starting to think I have some kind of dependency mismatch or I've setup my entry point wrong

app.json:
![image](https://user-images.githubusercontent.com/29917195/131363281-790cfd50-3543-4151-8434-20b7554cb87c.png)

package.json:
![image](https://user-images.githubusercontent.com/29917195/131363641-a49012ce-8aea-46be-8afa-738dea0ead18.png)

I'm not sure what the default entry is, but I set mine to be src/index.js, this contains all the providers including the NavigationContainer and any other dependencies like "react-native-gesture-handler"

**Comment by AlexMachin1997** [comment, marked as answer]
@raajnadar I asked on Twitter at it seems the problem was nothing to do navigating.

The actual problem was to do with one component I created. It was using the View component, I was passing strings to the margin properties when I should have only been passing numbers e.g. marginRight={10} as opposed to marginRight="10px" (Would be fine with styled-components).
